CRM/Report/Form.php

```diff
@@ -4779,6 +4779,16 @@ protected function setOutputMode() {
       FALSE,
       CRM_Utils_Array::value('task', $this->_params)
     ));
+    if ($this->_outputMode !== NULL) {
+      // We didn't get it using output, so we now try with task.
+      $this->_outputMode = str_replace('report_instance.', '', CRM_Utils_Request::retrieve(
+          'task',
+          'String',
+          CRM_Core_DAO::$_nullObject,
+          FALSE,
+          CRM_Utils_Array::value('task', $this->_params)
+        ));
+    }
```

Comment:
As `Save` and `Save a Copy` task has been removed from this PR, I was concerned about the remaining three actions(print, export, pdf) which seemed to work fine for me without this part. Can you please check and update? 
`Add Contact to Group` option disappearing is also an expected behaviour as per [above discussion](https://github.com/civicrm/civicrm-core/pull/9202/files#r90634203) .